compiler/src/yul/mappers/expressions.rs

```diff
@@ -62,6 +62,48 @@ pub fn expr(
     }
 }
 
+
+pub fn expr_bin_operation<'a>(scope: Shared<FunctionScope>, exp: &Spanned<vyp::Expr<'a>>) -> Result<ExtExpression, CompileError> {
+    if let vyp::Expr::BinOperation{ left, op, right } = &exp.node {
+        let yul_left = expr(Rc::clone(&scope), left)?.expression;
+        let yul_right = expr(Rc::clone(&scope), right)?.expression;
+
+        return match op.node {
+            vyp::BinOperator::Add => {
+                Ok(ExtExpression {
+                    expression: expression! { add([yul_left], [yul_right]) },
+                    location: Location::Value,
+                    typ: Type::Base(Base::U256),
+                })
+            },
+            vyp::BinOperator::Sub => {
+                Ok(ExtExpression {
+                    expression: expression! { sub([yul_left], [yul_right]) },
+                    location: Location::Value,
+                    typ: Type::Base(Base::U256),
+                })
+            }
+            vyp::BinOperator::Mult => {
+                Ok(ExtExpression {
+                    expression: expression! { mul([yul_left], [yul_right]) },
+                    location: Location::Value,
+                    typ: Type::Base(Base::U256),
+                })
+            },
+            vyp::BinOperator::Div => {
```

Comment:
@g-r-a-n-t  You mentioned in #24 that you would separate each arm into its own function. I was wondering if there was a strong argument in favor for that (beside just separation of concerns) because I believe it might just unnecessary bloat loc here. I can be convinced but wanted to here the reasoning :)

Reply:
I'm ok with this. Also, these functions will get even lighter with the semantic analysis pass.